controllers/resource/resource_generator.go

```diff
@@ -369,3 +358,143 @@ func (r *resourceGenerator) CreateDoguSecret(doguResource *k8sv1.Dogu, stringDat
+	p.specVolumes = volumes
+	return p
+}
+
+func (p *podSpecBuilder) enableServiceLinks(enable bool) *podSpecBuilder {
+	p.specEnableServiceLinks = enable
+	return p
+}
+
+func (p *podSpecBuilder) initContainers(initContainers ...*corev1.Container) *podSpecBuilder {
+	for _, initContainer := range initContainers {
+		if initContainer == nil {
+			continue
+		}
+
+		foundContainer := *initContainer
+		p.specInitContainers = append(p.specInitContainers, foundContainer)
+	}
+	return p
+}
+
+func (p *podSpecBuilder) containerCommandAndArgs(command, args []string) *podSpecBuilder {
+	p.specContainerCommand = command
+	p.specContainerArgs = args
+	return p
+}
+
+func (p *podSpecBuilder) containerLivenessProbe() *podSpecBuilder {
+	p.specContainerLivenessProbe = createLivenessProbe(p.theDogu)
+	return p
+}
+
+func (p *podSpecBuilder) containerStartupProbe() *podSpecBuilder {
+	p.specContainerStartupProbe = CreateStartupProbe(p.theDogu)
+	return p
+}
+
+func (p *podSpecBuilder) containerPullPolicy() *podSpecBuilder {
+	pullPolicy := corev1.PullIfNotPresent
+	if config.Stage == config.StageDevelopment {
+		pullPolicy = corev1.PullAlways
+	}
+
+	p.specContainerImagePullPolicy = pullPolicy
+
+	return p
+}
+
+func (p *podSpecBuilder) containerVolumeMounts(volumeMounts []corev1.VolumeMount) *podSpecBuilder {
+	p.specContainerVolumeMounts = volumeMounts
+	return p
+}
+
+func (p *podSpecBuilder) containerEnvVars(envVars []corev1.EnvVar) *podSpecBuilder {
+	p.specContainerEnvVars = envVars
+	return p
+}
+
+func (p *podSpecBuilder) containerResourceRequirements(reqs corev1.ResourceRequirements) *podSpecBuilder {
+	p.specContainerResourcesReq = reqs
+	return p
+}
+
+func (p *podSpecBuilder) build() *corev1.PodTemplateSpec {
+	result := &corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: p.metaAllLabels,
+		},
+		Spec: corev1.PodSpec{
+			ImagePullSecrets:   []corev1.LocalObjectReference{{Name: "k8s-dogu-operator-docker-registry"}},
+			Hostname:           p.theDoguResource.Name,
+			HostAliases:        p.specHostAliases,
+			Volumes:            p.specVolumes,
+			EnableServiceLinks: &p.specEnableServiceLinks,
+			ServiceAccountName: p.specServiceAccountName,
+			InitContainers:     p.specInitContainers,
+			Containers: []corev1.Container{{
+				Name:            p.theDoguResource.Name,
+				Image:           p.theDogu.Image + ":" + p.theDogu.Version,
+				Command:         p.specContainerCommand,
+				Args:            p.specContainerArgs,
+				LivenessProbe:   p.specContainerLivenessProbe,
+				StartupProbe:    p.specContainerStartupProbe,
+				ImagePullPolicy: p.specContainerImagePullPolicy,
+				VolumeMounts:    p.specContainerVolumeMounts,
+				Env:             p.specContainerEnvVars,
+				Resources:       p.specContainerResourcesReq,
+			}},
+		},
+	}
+
+	return result
+}
+
+func (p *podSpecBuilder) serviceAccount() *podSpecBuilder {
+	accountName, ok := getKubernetesServiceAccount(p.theDogu)
```

Comment:
add `serviceAccountName` as Parameter instead of nested function-call

Reply:
done.